modules/common/condition/funcs.go

```diff
@@ -86,6 +86,21 @@ func (conditions *Conditions) Set(c *Condition) {
 	conditions.Sort()
 }
 
+// Remove a condition from the slice of conditions
+func (conditions *Conditions) Remove(conditionType Type) {
+	if conditions == nil || len(*conditions) == 0 {
```

Comment:
might also want to use c.Has() to check if the type is in the conditions list and return if not, but then we'd run through them twice, once for checking and bellow to remove it.

Reply:
an option would be to change c.Has() to also return the index of the found condition and then delete the index in Remove(), but this would be a breaking change and would require updates to operators.

Reply:
I think c.Has() can be kept. and a new c.IndexOf() can be added that returns -1 or the index of the item. But it can be done independently.